### ISSUE TITLE
Can I use a crawler and ScrapeGraphAI together?

### ISSUE THREAD

**Issue by davideuler** [ScrapeGraphAI/Scrapegraph-ai#753]
**Is your feature request related to a problem? Please describe.**
To use a crawler like scrappy or crawlee with ScrapeGraphaAI together.
The crawler is responsible for crawl all the website, and filtering some contents.  For example get the pages under the same path with the root HTTP url. Another example is I need the pages with a specified regex pattern of the URL or of the content in the page.
And ScrapeGraphAI works as the content processing and analyzing.

**Describe the solution you'd like**
Solution 1: Graph API support passing just HTML content of webpage.

**Describe alternatives you've considered**
Solution 2: Implement something like content filtering, webpage filtering in the DepthSearchGraph. Hooks are also possible solutions.

**Additional context**
I want to crawl all product items in a shopping website. I want to crawl all mp3 file in a music website under a specify category like "Country Music".

**Comment by VinciGit00** [comment]
yes you have to fetch the code by yourself and add it with documentscrapergraph
